src/routing/utils.ts

```diff
@@ -105,3 +105,13 @@ export function getSortOrder(value: string | null | undefined, defaultValue: 'de
 
   return value as 'desc' | 'asc'
 }
+
+export function injectScript(url: string) {
+  const script = document.createElement('script')
+
+  script.src = url
+  script.type = 'text/javascript'
+  script.async = true
+
+  document.body.appendChild(script)
+}
```

Comment:
Can we, by any chance, make this function return a promise when the script finished loading?
The idea is to use this on the component that loads the scene editor so it gets loaded only when it's needed, improving the load times of the site. We would need to use the promise to show a loader before loading the scene editor and we'll need to set up a flag somewhere to know that the script ha already finished loading.

Reply:
@LautaroPetaccio -- thanks for the feedback! sure, we could return a promise when the script finished loading.

great idea! let me move the load script logic to the scene editor.